### PR TITLE
[#201] Connect Frontend Post Comments with Backend 

### LINKED ISSUE #201: Connect Frontend Post Comments with Backend
**Is your feature request related to a problem? Please describe.**
Commenting on posts is an existing feature that is already implemented, but the frontend and backend are not connected. 

This issue will focus on connecting the frontend post comments with backend to make post commenting fully functional.  

**Describe the solution you'd like**
Call the API which connects frontend posts with backend and test that post commenting is functional on the Post Detail Page. 

**Describe alternatives you've considered**
N/A

**Additional context**
This issue was discussed and allocated during the frontend team meeting on 29/03/2022. 

This is an example of post comments from the original mock-ups:

### PR DESCRIPTION
**Pull request by shreyasingaraju**
# Description:
Implemented post commenting by connecting frontend post comments with backend. This implementation creates a new post for each comment made, with the original post set as the 'parent'.  

fixes #201 

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](../CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project
- [x] My code has been commented
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

**Review by dogeliness — APPROVED** [review]
Looks good

**Review by scorpionknifes — CHANGES_REQUESTED** [review]
newTags and postTags are implemented in the backend.

Check https://github.com/SE701Team2/Updog/blob/main/backend/__tests__/posts.js#L187-L195 for detail
```js
const response = await request(server)
	  .post('/api/posts')
	  .set('Authorization', `Bearer ${authToken}`)
	  .send({
		    text_content: 'some random text 3',
		    parent: null,
		    tagIds: [tag1.id, tag2.id],
		    newTags: ['Cats', 'Dogs'],
	})
```

File: frontend/src/components/posts/comment/CommentController.jsx [hunk 1]
```diff
@@ -16,24 +18,29 @@ const CommentController = ({ postData }) => {
...
+          text_content: postText.replaceAll(/<.*?>/g, ''),
+        })
 
-        // // navigate to the newly made post
-        // setLoading(false)
-        // navigate(`/comments/${response.data.id}`)
+        // navigate to the newly made post comment
+        setLoading(false)
+        navigate(`/post/${postData.id}`)
+        window.location.reload()
```
Comment (scorpionknifes) [inline frontend/src/components/posts/comment/CommentController.jsx]: do we need a reload here?

File: frontend/src/components/posts/comment/CommentController.jsx [hunk 2]
```diff
@@ -16,24 +18,29 @@ const CommentController = ({ postData }) => {
   const { tags } = useContext(TagContext)
   const { handles } = useContext(HandleContext)
 
+  const navigate = useNavigate()
+
   const submitForm = async () => {
     if (postText) {
       try {
         setLoading(true)
+        console.log(postData)
```
Comment (scorpionknifes) [inline frontend/src/components/posts/comment/CommentController.jsx]:
```suggestion
```

**Review by dlee432 — APPROVED** [review]
LGTM

**Review by scorpionknifes — APPROVED** [review]
awesome work

**Pull request merged**